crates/legacy/src/service.rs

```diff
@@ -1044,13 +1082,34 @@ impl<Types: NodeType> AcceptsTxnSubmits<Types> for ProxyGlobalState<Types> {
             txns.len(),
             txns.iter().map(|txn| txn.commit()).collect::<Vec<_>>()
         );
+        for txn in txns.clone() {
+            let _ = self
+                .global_state
+                .write_arc()
+                .await
+                .set_tx_status(txn.commit(), TransactionStatus::Pending);
+        }
         let response = self
             .global_state
             .read_arc()
             .await
-            .submit_client_txns(txns)
+            .submit_client_txns(txns.clone())
             .await;
 
+        let pairs: Vec<(<Types as NodeType>::Transaction, Result<_, _>)> = (0..txns.len())
+            .map(|i| (txns[i].clone(), response[i].clone()))
```

Comment:
Cloning the transaction here is a bit wasteful, as we're only interested in the hash.

Reply:
good point!